tests/llama_index/test_llama_index_model_export.py

```diff
@@ -0,0 +1,259 @@
+import numpy as np
+import pandas as pd
+import pytest
+from llama_index.core import QueryBundle
+from llama_index.core.base.response.schema import (
+    Response,
+)
+from llama_index.core.chat_engine.types import (
+    AgentChatResponse,
+)
+from llama_index.core.llms import ChatMessage
+
+import mlflow
+import mlflow.llama_index
+import mlflow.pyfunc
+from mlflow.llama_index.pyfunc_wrapper import create_engine_wrapper
+from mlflow.models import infer_signature
+
+
+@pytest.mark.parametrize(
+    "index_fixture",
+    [
+        "single_index",
+        "multi_index",
+        "single_graph",
+    ],
+)
+def test_llama_index_native_save_and_load_model(request, index_fixture, tmp_path):
+    index = request.getfixturevalue(index_fixture)
+    mlflow.llama_index.save_model(index, tmp_path, engine_type="query")
+    loaded_model = mlflow.llama_index.load_model(tmp_path)
+
+    assert type(loaded_model) == type(index)
+    assert loaded_model.as_query_engine().query("Spell llamaindex").response.lower() != ""
+
+
+@pytest.mark.parametrize(
+    "index_fixture",
+    [
+        "single_index",
+        "multi_index",
+        "single_graph",
+    ],
+)
+def test_llama_index_native_log_and_load_model(request, index_fixture):
+    index = request.getfixturevalue(index_fixture)
+    with mlflow.start_run():
+        logged_model = mlflow.llama_index.log_model(index, "model", engine_type="query")
+
+    loaded_model = mlflow.llama_index.load_model(logged_model.model_uri)
+
+    assert "llama_index" in logged_model.flavors
+    assert type(loaded_model) == type(index)
+    engine = loaded_model.as_query_engine()
+    assert engine is not None
+    assert engine.query("Spell llamaindex").response != ""
+
+
+@pytest.mark.parametrize(
+    "engine_type",
+    ["query", "retriever"],
+)
+def test_format_predict_input_correct(single_index, engine_type):
```

Comment:
nit: are these `_format_predict_input` tests supposed to be in the pyfunc wrapper test file?